conan/internal/api/detect/detect_api.py

```diff
@@ -461,14 +461,11 @@ def detect_gcc_compiler(compiler_exe="gcc"):
             if "clang" in out:
                 return None, None, None
 
-        ret, out = detect_runner('%s -dumpversion' % compiler_exe)
+        ret, out = detect_runner('%s -dumpfullversion -dumpversion' % compiler_exe)
```

Comment:
Is it possible that some non mainstream gcc-like compilers do not support ``-dumpfullversion`` and crash?
I think we previously didn't add this because of the potential risk of breaking others, but we can try